### PR TITLE
Test on 3.10.0-beta.2

### PR DESCRIPTION
**Pull request by zariiii9003**
Pytest update was necessary due to https://github.com/pytest-dev/pytest/pull/8540

<details><summary>Trace</summary>
<p>

```
GLOB sdist-make: /home/runner/work/python-can/python-can/setup.py
gh create: /home/runner/work/python-can/python-can/.tox/gh
gh installdeps: pytest~=5.3, pytest-timeout~=1.3, pytest-cov~=2.8, coverage==4.5.*, codecov==2.1.10, hypothesis~=4.56, pyserial~=3.0
gh inst: /home/runner/work/python-can/python-can/.tox/.tmp/package/1/python-can-4.0.0.dev2.zip
gh installed: attrs==21.2.0,certifi==2021.5.30,chardet==4.0.0,codecov==2.1.10,coverage==4.5.4,hypothesis==4.57.1,idna==2.10,more-itertools==8.8.0,msgpack==1.0.2,mypy-extensions==0.4.3,packaging==20.9,pluggy==0.13.1,py==1.10.0,pyparsing==2.4.7,pyserial==3.5,pytest==5.4.3,pytest-cov==2.10.1,pytest-timeout==1.4.2,python-can @ file:///home/runner/work/python-can/python-can/.tox/.tmp/package/1/python-can-4.0.0.dev2.zip,requests==2.25.1,sortedcontainers==2.4.0,urllib3==1.26.5,wcwidth==0.2.5,wrapt==1.12.1
gh run-test-pre: PYTHONHASHSEED='4019826288'
gh run-test: commands[0] | pytest
Traceback (most recent call last):
  File "/home/runner/work/python-can/python-can/.tox/gh/bin/pytest", line 8, in <module>
    sys.exit(main())
  File "/home/runner/work/python-can/python-can/.tox/gh/lib/python3.10/site-packages/_pytest/config/__init__.py", line 105, in main
    config = _prepareconfig(args, plugins)
  File "/home/runner/work/python-can/python-can/.tox/gh/lib/python3.10/site-packages/_pytest/config/__init__.py", line 257, in _prepareconfig
    return pluginmanager.hook.pytest_cmdline_parse(
  File "/home/runner/work/python-can/python-can/.tox/gh/lib/python3.10/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/runner/work/python-can/python-can/.tox/gh/lib/python3.10/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/runner/work/python-can/python-can/.tox/gh/lib/python3.10/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/home/runner/work/python-can/python-can/.tox/gh/lib/python3.10/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/home/runner/work/python-can/python-can/.tox/gh/lib/python3.10/site-packages/_pytest/helpconfig.py", line 90, in pytest_cmdline_parse
    config = outcome.get_result()
  File "/home/runner/work/python-can/python-can/.tox/gh/lib/python3.10/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/runner/work/python-can/python-can/.tox/gh/lib/python3.10/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/runner/work/python-can/python-can/.tox/gh/lib/python3.10/site-packages/_pytest/config/__init__.py", line 836, in pytest_cmdline_parse
    self.parse(args)
  File "/home/runner/work/python-can/python-can/.tox/gh/lib/python3.10/site-packages/_pytest/config/__init__.py", line 1044, in parse
    self._preparse(args, addopts=addopts)
  File "/home/runner/work/python-can/python-can/.tox/gh/lib/python3.10/site-packages/_pytest/config/__init__.py", line 992, in _preparse
    self.pluginmanager.load_setuptools_entrypoints("pytest11")
  File "/home/runner/work/python-can/python-can/.tox/gh/lib/python3.10/site-packages/pluggy/manager.py", line 299, in load_setuptools_entrypoints
    plugin = ep.load()
  File "/opt/hostedtoolcache/Python/3.10.0-beta.2/x64/lib/python3.10/importlib/metadata/__init__.py", line 162, in load
    module = import_module(match.group('module'))
  File "/opt/hostedtoolcache/Python/3.10.0-beta.2/x64/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "/home/runner/work/python-can/python-can/.tox/gh/lib/python3.10/site-packages/_pytest/assertion/rewrite.py", line 143, in exec_module
    source_stat, co = _rewrite_test(fn, self.config)
  File "/home/runner/work/python-can/python-can/.tox/gh/lib/python3.10/site-packages/_pytest/assertion/rewrite.py", line 330, in _rewrite_test
    co = compile(tree, fn, "exec", dont_inherit=True)
TypeError: required field "lineno" missing from alias
ERROR: InvocationError for command /home/runner/work/python-can/python-can/.tox/gh/bin/pytest (exited with code 1)
```

</p>
</details>

### CODE REVIEW

**Review by felixdivo — APPROVED** [review]

**Comment by zariiii9003** [comment]
@mergifyio rebase

**Comment by zariiii9003** [comment]
I tried to fix the codecov issues by using the codecov action, but maybe the problems are not even caused by my changes: https://github.com/codecov/codecov-action/issues/330

**Comment by felixdivo** [comment]
This would also remove coverage collection when testing on Travis (which is only relevant for Socketcan).

**Comment by zariiii9003** [comment]
> This would also remove coverage collection when testing on Travis (which is only relevant for Socketcan).

I don't think so. For travis it is still called via tox file.

@hardbyte the  replies on stackoverflow suggest that rotating the codecov token might help. Could you try?

**Comment by felixdivo** [comment]
> For travis it is still called via tox file.

Yes and No. If I understand it correctly, that is removed from the tox file [here](https://github.com/hardbyte/python-can/pull/1069/files#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449L26).

**Comment by zariiii9003** [comment]
> > For travis it is still called via tox file.
> 
> Yes and No. If I understand it correctly, that is removed from the tox file [here](https://github.com/hardbyte/python-can/pull/1069/files#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449L26).

But only for the github environment. The command is still present for Travis

**Comment by felixdivo** [comment]
The failures of Travis should be fixed on develop by now, so an update should get this CI to become green:

@Mergifyio update

**Review by hardbyte — APPROVED** [review]
Sorry for the delay looking at this. 

I assume the codecov token is fine as is?

**Review by felixdivo — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
         experimental: [false]
         python-version: [3.6, 3.7, 3.8, 3.9, pypy3]
         include:
-          - python-version: 3.10.0-alpha.7  # Newest: https://github.com/actions/python-versions/blob/main/versions-manifest.json
+          - python-version: 3.10.0-beta.2  # Newest: https://github.com/actions/python-versions/blob/main/versions-manifest.json
```
Comment (felixdivo) [inline .github/workflows/build.yml:18]:
```suggestion
          - python-version: 3.10.0-beta.4  # Newest: https://github.com/actions/python-versions/blob/main/versions-manifest.json
```

**Pull request merged**